### PR TITLE
fix(node-fetch): use stream.Readable instead of web streams

### PR DESCRIPTION
**Pull request by paperclover**
### What does this PR do?

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [X] Code changes

### How did you verify your code works?

New tests

### CODE REVIEW

File: src/js/thirdparty/node-fetch.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    if (body instanceof Readable) {
+      for await (const chunk of body) {
+        chunks.push(chunk);
+      }
+      init = { ...init, body: new Blob(chunks) };
+    }
+  }
+
+  const response = await nativeFetch(url, init);
+  Object.setPrototypeOf(response, Response.prototype);
```
Comment (paperclover) [inline src/js/thirdparty/node-fetch.ts:39]: how slow is this prototype switch? should this be a subclass that implements all the getters or something else?

A good thing about this is it makes the wrapper very minimal in terms of # of js code.

Comment (Jarred-Sumner) [reply src/js/thirdparty/node-fetch.ts:39]: pretty bad but not as bad as the network call

File: src/js/node/stream.js [hunk 2]
```diff
@@ -3327,7 +3327,7 @@ var require_readable = __commonJS({
 
         streamReadable.pause();
 
-        const cleanup = finished(streamReadable, error => {
+        const cleanup = eos(streamReadable, error => {
```
Comment (Jarred-Sumner) [inline src/js/node/stream.js:3330]: How intentional is this on a scale of 1 to 10

**Review by Jarred-Sumner — APPROVED** [review]

File: src/js/node/stream.js (hunk 2 above)
Comment (paperclover) [reply src/js/node/stream.js:3330]: i looked at the place we copied this from
```
import { eos as finished } from ...
```

(there is no variable named `finished` in this scope)

so 10

**Comment by paperclover** [comment]
rebased, letting tests run.

**Pull request merged**